.github/workflows/publish.yaml

```diff
@@ -0,0 +1,31 @@
+name: Publish
+
+on:
+  push:
+    branches:
+      - master
+
+jobs:
+  publish:
+    name: Publish to NPM
+    runs-on: ubuntu-latest
+    steps:
+      -
+        uses: actions/checkout@v4
+      -
+        uses: actions/setup-node@v4
+        with:
+          node-version: 20
+          registry-url: https://registry.npmjs.org/
+      -
+        name: Install Dependencies
+        run: npm install
+      -
+        name: Publish to NPM
+        run: NODE_AUTH_TOKEN=${{secrets.NPM_TOKEN}} npm publish --tag latest --access public
+      -
+        name: Tag Version
+        run: |
+          git_tag=v`node -p "require('./package.json').version"`
+          git tag $git_tag
+          git push origin $git_tag
```

Comment:
the bumping should be done manually, yes.

Reply:
I really dislike that, the new release process we are adopting these days don't suffer from this unnecessary manual step, why should we deal with this manually here? Basically all our public repos (except CLI which is in progress) don't requite this now, we release via workflow dispatch and bump based on conventional commits with possible manual override. We could adopt the same in here if you want, I'd let @janbuchar to make it work since its all his work. It also means automated changelogs and GH releases and other goodies.

Reply:
Ok, nice. Please send me some PR/commit that implements it, I'll do it here. Thanks!

Reply:
One from the CLI, we have also canary builds in there, I don't know what we want from this repo, we could as well just ship to stable like we do in apify-shared-js, but if we want beta versions and stable releases via workflow dispatch, that's what the CLI and client are doing.

https://github.com/apify/apify-cli/pull/691

In client we had two PRs since we missed one thing

https://github.com/apify/apify-client-js/pull/596
https://github.com/apify/apify-client-js/pull/605